### PR TITLE
FIX-BESTPRACTICES: data-factory-v2-azure-sql-database-to-sql-data-warehouse-copy

### PR DESCRIPTION
**Pull request by sameera-spektra**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: quickstarts/microsoft.datafactory/data-factory-v2-azure-sql-database-to-sql-data-warehouse-copy/azuredeploy.json [hunk 1]
```diff
@@ -32,6 +39,12 @@
       "metadata": {
         "description": "Connection string for the Azure SQL Data Warehouse"
       }
+    },
+    "databaseAdminPassword": {
+      "type": "string",
```
Comment (bmoore-msft) [inline quickstarts/microsoft.datafactory/data-factory-v2-azure-sql-database-to-sql-data-warehouse-copy/azuredeploy.json:44]: must be secureString

File: quickstarts/microsoft.datafactory/data-factory-v2-azure-sql-database-to-sql-data-warehouse-copy/prereqs/prereq.azuredeploy.parameters.json [hunk 2]
```diff
@@ -6,7 +6,10 @@
             "value": "GEN-UNIQUE"
         },
         "databaseAdminPassword": {
-            "value": "GEN-PASSWORD"
+            "value": "GEN-SELFSIGNED-CERT-PASSWORD"
+        },
+        "sqlServerBase": {
```
Comment (bmoore-msft) [inline quickstarts/microsoft.datafactory/data-factory-v2-azure-sql-database-to-sql-data-warehouse-copy/prereqs/prereq.azuredeploy.parameters.json:11]: remove this param and use the environment function to get the endpoint

File: quickstarts/microsoft.datafactory/data-factory-v2-azure-sql-database-to-sql-data-warehouse-copy/prereqs/prereq.azuredeploy.json [hunk 3]
```diff
@@ -110,25 +115,28 @@
...
+    },
+    "storageAccountResourceId": {
       "type": "string",
-      "value": "[concat('DefaultEndpointsProtocol=https;AccountName=',variables('storageAccountName'),';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName')), '2016-01-01').keys[0].value)]"
+      "value": "[resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName'))]"
     },
     "sqlDatabaseCS": {
       "type": "string",
-      "value": "[concat('Server=tcp:', variables('serverName'),'.database.windows.net,1433;Initial Catalog=',variables('databaseName'), ';Persist Security Info=False;User ID=', parameters('databaseAdminUsername'), ';Password=', parameters('databaseAdminPassword'), ';Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;')]"
+      "value": "[concat('Server=tcp:', variables('serverName'), parameters('sqlServerBase'), ',1433;Initial Catalog=',variables('databaseName'), ';Persist Security Info=False;User ID=', parameters('databaseAdminUsername'))]"
```
Comment (bmoore-msft) [inline quickstarts/microsoft.datafactory/data-factory-v2-azure-sql-database-to-sql-data-warehouse-copy/prereqs/prereq.azuredeploy.json:135]: this is a really strange pattern - we're outputting part of a connectionString - which is usually a secret.  So it looks like we have a parameter in the template which is a secret but is not a secureString.  I think we have a few options
1) rename this output (and the param) to make it clear it's not a secret or 
1) construct then entire connstr in the main template (by passing in all the parts needed to construct it
1) make it a secret and store it in keyVault (and then retrieve it from there in the template

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.datafactory/data-factory-v2-azure-sql-database-to-sql-data-warehouse-copy/azuredeploy.json (hunk 1 above)
Comment (sameera-spektra) [reply quickstarts/microsoft.datafactory/data-factory-v2-azure-sql-database-to-sql-data-warehouse-copy/azuredeploy.json:44]: Fixed

File: quickstarts/microsoft.datafactory/data-factory-v2-azure-sql-database-to-sql-data-warehouse-copy/prereqs/prereq.azuredeploy.parameters.json (hunk 2 above)
Comment (sameera-spektra) [reply quickstarts/microsoft.datafactory/data-factory-v2-azure-sql-database-to-sql-data-warehouse-copy/prereqs/prereq.azuredeploy.parameters.json:11]: Fixed

File: quickstarts/microsoft.datafactory/data-factory-v2-azure-sql-database-to-sql-data-warehouse-copy/prereqs/prereq.azuredeploy.json (hunk 3 above)
Comment (sameera-spektra) [reply quickstarts/microsoft.datafactory/data-factory-v2-azure-sql-database-to-sql-data-warehouse-copy/prereqs/prereq.azuredeploy.json]: Updated with option 2

**Pull request merged**